File: maps/src/maps/map_layer.cpp
#include <maps/map_layer.h>

using namespace maps;

MapLayer::MapLayer(MapLayerType layer_type, const std::string& layer_name)
  : layer_type_(layer_type), layer_name_(layer_name)
{
}

MapLayerType MapLayer::getLayerType() const
{
  return layer_type_;
}

const std::string& MapLayer::getLayerName() const
{
  return layer_name_;
}

File: maps/src/maps/map_layer.py
from enum import Enum


class MapType(Enum):
    LANE = "LANE"
    LANE_ANNOTATION = "LANE_ANNOTATION"
    ROAD = "ROAD"
    MAP_READER = "MAP_READER"
    FREE_SPACE = "FREE_SPACE"
    RADAR_ZONE = "RADAR_ZONE"
    LOCALIZATION_ZONE = "LOCALIZATION_ZONE"
    DISENGAGE_ZONE = "DISENGAGE_ZONE"


class MapLayer(object):
    def __init__(self, layer_type, layer_name=''):
        self.layer_type = layer_type
        self.layer_name = layer_name

    # TODO: Open Questions:
    #  Do we want the generic get_feature / get_features method here? This would essentially become SingleMapLayer
    #  Restrict all map layers to be geojson user facing?
